bee-message/src/parent.rs

```diff
@@ -55,34 +65,18 @@ impl Parents {
 }
 
 impl Packable for Parents {
-    type Error = Error;
+    type UnpackError = Error;
 
-    fn packed_len(&self) -> usize {
-        0u8.packed_len() + self.len() * MessageId::LENGTH
+    fn pack<P: Packer>(&self, packer: &mut P) -> Result<(), P::Error> {
+        self.0.pack(packer)
     }
 
-    fn pack<W: Write>(&self, writer: &mut W) -> Result<(), Self::Error> {
-        (self.len() as u8).pack(writer)?;
-
-        for parent in self.iter() {
-            parent.pack(writer)?;
-        }
-
-        Ok(())
-    }
-
-    fn unpack_inner<R: Read + ?Sized, const CHECK: bool>(reader: &mut R) -> Result<Self, Self::Error> {
-        let parents_len = u8::unpack_inner::<R, CHECK>(reader)? as usize;
-
-        if CHECK && !Parents::COUNT_RANGE.contains(&parents_len) {
-            return Err(Error::InvalidParentsCount(parents_len));
-        }
-
-        let mut inner = Vec::with_capacity(parents_len);
-        for _ in 0..parents_len {
-            inner.push(MessageId::unpack_inner::<R, CHECK>(reader)?);
-        }
+    fn unpack<U: Unpacker, const VERIFY: bool>(
+        unpacker: &mut U,
+    ) -> Result<Self, UnpackError<Self::UnpackError, U::Error>> {
+        let inner = BoxedSlicePrefix::<MessageId, ParentCount>::unpack::<_, VERIFY>(unpacker)
+            .map_packable_err(|err| Error::InvalidParentCount(err.into_prefix().into()))?;
 
-        Self::new(inner)
+        Self::from_boxed_slice::<VERIFY>(inner).map_err(UnpackError::Packable)
```

Comment:
Maybe we can get rid of the impl with verify_with ?

Reply:
yeah we can! done!